ws-services/src/main/java/org/egov/waterConnection/service/EnrichmentService.java

```diff
@@ -0,0 +1,133 @@
+package org.egov.waterConnection.service;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.tracer.model.CustomException;
+import org.egov.waterConnection.config.WSConfiguration;
+import org.egov.waterConnection.model.Property;
+import org.egov.waterConnection.model.WaterConnection;
+import org.egov.waterConnection.model.WaterConnectionRequest;
+import org.egov.waterConnection.model.Idgen.IdResponse;
+import org.egov.waterConnection.repository.IdGenRepository;
+import org.egov.waterConnection.model.SearchCriteria;
+import org.egov.waterConnection.util.WaterServicesUtil;
+import org.egov.waterConnection.validator.ValidateProperty;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+@Service
+public class EnrichmentService {
+
+	@Autowired
+	WaterServicesUtil waterServicesUtil;
+
+	@Autowired
+	IdGenRepository idGenRepository;
+
+	@Autowired
+	WSConfiguration config;
+
+	@Autowired
+	ValidateProperty validateProperty;
+
+	/**
+	 * 
+	 * @param waterConnectionList
+	 *            List of water connection for enriching the water connection
+	 *            with property.
+	 * @param requestInfo
+	 *            is RequestInfo from request
+	 */
+	public void enrichWaterSearch(List<WaterConnection> waterConnectionList, RequestInfo requestInfo,
+			SearchCriteria waterConnectionSearchCriteria) {
+		waterConnectionList.forEach(waterConnection -> {
```

Comment:
If this is actually going to be used with List of water connections, then do refactor this to use a single property search call instead of doing this in a loop